File: CS50x - 2024/testes_C/mario.c
#include <stdio.h>
#include <cs50.h>

void print_space(int espaço);
void print_row(int largura);

int main(void)
{
    int altura;
    do
    {
        altura = get_int("Altura: ");
    }
    while(altura < 1 || altura > 8);

    for (int i = 0; i < altura; i++)
    {
        print_space(altura - 1 - i);
        print_row(i + 1);
    }
}

void print_space(int espaço)
{
    for (int h = 0; h < espaço; h++)
    {
        printf(" ");
    }
}

void print_row(int largura)
{
    for (int j = 0; j < largura; j++)
    {
        printf("#");
    }
    printf("\n");
}

File: CS50x - 2024/sentimental-mario-less/mario.py
from cs50 import get_int

while True:
    altura = get_int("Height: ")
    if altura > 0 and altura < 9:
        break

for i in range(altura):
    print(" " * (altura - i - 1) + "#" * (i + 1), end='')
    print("  "+ "#" * (i + 1))
